folding-schemes/src/folding/nova/serialize.rs

```diff
@@ -0,0 +1,216 @@
+    }
+
+    fn serialized_size(&self, compress: ark_serialize::Compress) -> usize {
+        self.cs_params.serialized_size(compress)
+            + self.cf_cs_params.serialized_size(compress)
+            + self.i.serialized_size(compress)
+            + self.z_0.serialized_size(compress)
+            + self.z_i.serialized_size(compress)
+            + self.w_i.serialized_size(compress)
+            + self.u_i.serialized_size(compress)
+            + self.W_i.serialized_size(compress)
+            + self.U_i.serialized_size(compress)
+            + self.cf_W_i.serialized_size(compress)
+            + self.cf_U_i.serialized_size(compress)
+            + self.r1cs.serialized_size(compress)
+            + self.cf_r1cs.serialized_size(compress)
+            + self.poseidon_config.full_rounds.serialized_size(compress)
+            + self
+                .poseidon_config
+                .partial_rounds
+                .serialized_size(compress)
+            + self.poseidon_config.alpha.serialized_size(compress)
+            + self.poseidon_config.ark.serialized_size(compress)
+            + self.poseidon_config.mds.serialized_size(compress)
+            + self.poseidon_config.rate.serialized_size(compress)
+            + self.poseidon_config.capacity.serialized_size(compress)
+    }
+
+    fn serialize_compressed<W: Write>(
+        &self,
+        writer: W,
+    ) -> Result<(), ark_serialize::SerializationError> {
+        self.serialize_with_mode(writer, ark_serialize::Compress::Yes)
+    }
+
+    fn compressed_size(&self) -> usize {
+        self.serialized_size(ark_serialize::Compress::Yes)
+    }
+
+    fn serialize_uncompressed<W: Write>(
+        &self,
+        writer: W,
+    ) -> Result<(), ark_serialize::SerializationError> {
+        self.serialize_with_mode(writer, ark_serialize::Compress::No)
+    }
+
+    fn uncompressed_size(&self) -> usize {
+        self.serialized_size(ark_serialize::Compress::No)
+    }
+}
+
+// Note that we can't derive or implement `CanonicalDeserialize` directly.
+// This is because `CurveVar` notably does not implement the `Sync` trait.
+impl<C1, GC1, C2, GC2, FC, CS1, CS2> Nova<C1, GC1, C2, GC2, FC, CS1, CS2>
+where
+    C1: CurveGroup,
+    C2: CurveGroup,
+    FC: FCircuit<C1::ScalarField, Params = ()>,
+    CS1: CommitmentScheme<C1>,
+    CS2: CommitmentScheme<C2>,
+    <C1 as CurveGroup>::BaseField: PrimeField,
+    <C2 as CurveGroup>::BaseField: PrimeField,
+    <C1 as Group>::ScalarField: Absorb,
+    <C2 as Group>::ScalarField: Absorb,
+    C1: CurveGroup<BaseField = C2::ScalarField, ScalarField = C2::BaseField>,
+    for<'a> &'a GC1: GroupOpsBounds<'a, C1, GC1>,
+    for<'a> &'a GC2: GroupOpsBounds<'a, C2, GC2>,
+    GC1: CurveVar<C1, <C2 as Group>::ScalarField>,
+    GC1: ToConstraintFieldGadget<<C2 as Group>::ScalarField>,
+    GC2: CurveVar<C2, <C2 as CurveGroup>::BaseField>,
+    CS1::ProverParams: CanonicalDeserialize,
+    CS2::ProverParams: CanonicalDeserialize,
+{
+    #[allow(dead_code)]
```

Comment:
seems that this `#[allow(dead_code)]` is not needed?

Reply:
Had `cargo clippy` complaining in earlier commits, good catch!